app/services/sign_in/client_assertion_validator.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module SignIn
+  class ClientAssertionValidator
+    attr_reader :client_assertion, :client_assertion_type, :client_config
+
+    def initialize(client_assertion:, client_assertion_type:, client_config:)
+      @client_assertion = client_assertion
+      @client_assertion_type = client_assertion_type
+      @client_config = client_config
+    end
+
+    def perform
+      validate_client_assertion_type
+      validate_iss
+      validate_subject
+      validate_audience
+    end
+
+    private
+
+    def validate_client_assertion_type
+      if client_assertion_type != Constants::Auth::CLIENT_ASSERTION_TYPE
+        raise Errors::ClientAssertionTypeInvalidError.new message: 'Client assertion type is not valid'
+      end
+    end
+
+    def validate_iss
+      if decoded_client_assertion.iss != client_config.client_id
+        raise Errors::ClientAssertionAttributesError.new message: 'Client assertion issuer is not valid'
+      end
+    end
+
+    def validate_subject
+      if decoded_client_assertion.sub != client_config.client_id
+        raise Errors::ClientAssertionAttributesError.new message: 'Client assertion subject is not valid'
+      end
+    end
+
+    def validate_audience
+      unless decoded_client_assertion.aud.match(token_route)
+        raise Errors::ClientAssertionAttributesError.new message: 'Client assertion audience is not valid'
+      end
+    end
+
+    def token_route
+      "#{Settings.hostname}#{Constants::Auth::TOKEN_ROUTE_PATH}"
```

Comment:
This is missing a slash - 
```suggestion
      "#{Settings.hostname}/#{Constants::Auth::TOKEN_ROUTE_PATH}"
```